src/app/_services/jstor.redirect.service.ts

```diff
@@ -1,24 +1,56 @@
+import { gql } from '@apollo/client/core';
 import { Injectable } from '@angular/core';
 import { ActivatedRouteSnapshot, CanActivateChild } from "@angular/router";
+import { ApolloClient, InMemoryCache } from "@apollo/client/core";
+
+const FLAGS_QUERY = gql`
+  query AiwFlagList($flagsFlagList: [String]) {
+    flags(flagList: $flagsFlagList) {
+      enabled
+    }
+  }
+`;
+
+const APOLLO = new ApolloClient({
+  uri: "/ui/data-fetch/gateway",
+  headers: { authorization: "aiw-ui" },
+});
+
+const REDIRECT_FLAG = "artstor_client_redirect";
 
 @Injectable()
 export class JSTORRedirect implements CanActivateChild {
    canActivateChild(route: ActivatedRouteSnapshot) {
 
+    const options = {
+      query: FLAGS_QUERY,
+      variables: {
+        flagsFlagList: [REDIRECT_FLAG],
+      },
+    } as any;
+    const flags = await APOLLO.query(options);
+    const data = flags.response ? flags.response.data : null;
+    const enabledFlags = data ? data.flags.enabled : [];
+    const doRedirect = enabledFlags.includes(REDIRECT_FLAG);
+
     const currentRequest = window.location.href;
 
     if (currentRequest.includes('/#/')) {
       const params = new URLSearchParams({artstorPath: currentRequest }).toString();
       fetch(`/get-the-redirect-please/?${params}`).then(async resp => {
-        // TODO: Logic to handle "location" response from server and 301 user to that
         const data = await resp.json();
         if (data.location) {
```

Comment:
This is a little out of scope but do we know what we want to do if `data.location` evaluates to false? Because in that case, the user will continue to hang out on artstor? Do we eventually need to add a fallback?

Reply:
I think it is the server-side responsibility to do this, and don't want to get overly defensive here—TBD, I think.